src/flint/fmpz.jl

```diff
@@ -1078,6 +1078,20 @@ function sqrtmod(x::ZZRingElem, m::ZZRingElem)
   return z
 end
 
+function sqrtmod_pk(a::Int, p::Int, k::Int)
+  res = Ref(Ptr{UInt}())
+  n = @ccall libflint.n_sqrtmod_primepow(res::Ref{Ptr{UInt}}, a::Int, p::Int, k::Int)::Int
```

Comment:
The c-functions is for uint, so for negative a (and probably p) this will do funny things